### PR TITLE
add trivi scanning ci

### PR DESCRIPTION
**Pull request by johnoloughlin**

### CODE REVIEW

**Comment by johnoloughlin** [comment]
shouldn't this have triggered a job?

**Comment by Eoghan1232** [comment]
> shouldn't this have triggered a job?

I think a job must be added first, then follow up runs get run

**Review by Eoghan1232 — APPROVED** [review]

File: .github/workflows/trivy.yml [hunk 1]
```diff
@@ -0,0 +1,49 @@
+# This workflow uses actions that are not certified by GitHub.
+# They are provided by a third-party and are governed by
+# separate terms of service, privacy policy, and support
+# documentation.
+
+name: trivy
+
+on:
+  push:
+    branches: [ "master" ]
```
Comment (Eoghan1232) [inline .github/workflows/trivy.yml]: change to main

File: .github/workflows/trivy.yml [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+  pull_request:
+    # The branches below must be a subset of the branches above
+    branches: [ "master" ]
```
Comment (Eoghan1232) [inline .github/workflows/trivy.yml]: same here

**Review by Eoghan1232 — CHANGES_REQUESTED** [review]

File: .github/workflows/trivy.yml [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+      actions: read # only required for a private repository by github/codeql-action/upload-sarif to get the Action run status
+    name: Build
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@b4ffde65f46336ab88eb53be808477a3936bae11
+
+
+      - name: Build the Docker image
+        run: make docker-build IMAGE_REGISTRY=localhost:5000/ IMAGE_VERSION=latest
```
Comment (Eoghan1232) [inline .github/workflows/trivy.yml:34]: will the space cause an issue?

Comment (johnoloughlin) [reply .github/workflows/trivy.yml:34]: its a make env var

**Comment by Eoghan1232** [comment]
lgtm

**Review by Eoghan1232 — APPROVED** [review]

**Pull request merged**